backend/netstorage/netstorage.go

```diff
@@ -0,0 +1,1369 @@
+	//we return object updated with source stats
+	//we don't refetch the obj after upload
+	if err != nil {
+		fs.Debugf(nil, "in Update, netStorageUploadRequest returns err: %v", err)
+		return err
+	}
+	return nil
+}
+
+// newObjectWithInfo creates an fs.Object for any netstorage.File or symlink.
+// If it can't be found it returns the error fs.ErrorObjectNotFound.
+// It returns fs.ErrorNotAFile error for objects other than a file or a symlink
+// but still fills the fs.Object structure (for directory operations).
+func (f *Fs) newObjectWithInfo(remote string, info *File) (fs.Object, error) {
+	if info == nil {
+		return nil, fs.ErrorObjectNotFound
+	}
+	URL := f.url(remote)
+	var size int64
+	if info.Type == "symlink" {
+		// file size for symlinks is absent but for .rclonelink to work
+		// the size should be the length of the target name
+		size = int64(len(info.Target))
+	} else {
+		var err error
+		size, err = strconv.ParseInt(info.Size, 10, 64)
+		if err != nil && info.Type == "file" {
+			fs.Errorf(nil, "ERROR: can't parse int: %s for URL: %s", info.Size, URL)
+			return nil, fs.ErrorObjectNotFound // there is no better error code
+		}
+	}
+	o := &Object{
+		fs:       f,
+		filetype: info.Type,
+		remote:   remote,
+		size:     size,
+		modTime:  info.Mtime,
+		md5sum:   info.Md5,
+		fullURL:  URL,
+		target:   info.Target,
+	}
+	if info.Type == "dir" {
+		return o, fs.ErrorNotAFile
+	}
+	return o, nil
+}
+
+// wraps around Base for normal use case httprequests that have no need for the res object
+func (f *Fs) sendHTTPRequest(ctx context.Context, URL string, actionH string, httpType string, xmlbody ResponseObject) (ResponseObject, error) {
+	xml, res, err := f.sendHTTPRequestBase(ctx, URL, actionH, httpType, xmlbody, nil)
+	if err != nil {
+		return nil, err
+	}
+	fs.CheckClose(res.Body, &err)
+	return xml, err
+}
+
+// generic utility function to send and process HTTP CMS API requests
+func (f *Fs) sendHTTPRequestBase(ctx context.Context, URL string, actionH string, httpType string, xmlbody ResponseObject, options []fs.OpenOption) (ResponseObject, *http.Response, error) {
+	errInfo := httpType + " " + URL + " " + actionH
+	u, err := url.Parse(URL)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to Parse URL: "+errInfo)
+	}
+	req, err := http.NewRequestWithContext(ctx, httpType, URL, nil)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to NewRequest: "+errInfo)
+	}
+	f.addHeaders(req, u.RequestURI(), actionH)
+	if options != nil {
+		fs.OpenOptionAddHTTPHeaders(req.Header, options)
+		for x := 0; x < len(options); x++ {
+			fs.Debugf(nil, "requestbase options: %s for URL: %s", options[x].String(), URL)
+		}
+	}
+	var res *http.Response
+	//res is always saved to the most recent httpClient call
+	//retries if 429 or 500, doesn't retry if 404 or any other error
+	err = f.pacer.Call(func() (bool, error) {
+		fs.Debugf(nil, "CALLING httpClient Do: %s", URL)
+		res, err = f.httpClient.Do(req)
```

Comment:
I recommend using the abstractions in lib/http here - it looks like this could be replaced with a call to `rest.CallXML`

Take a look at how the webdav backend works, and take a look at the `shouldRetry` function.